src/library.cpp

```diff
@@ -160,13 +160,17 @@ void Library::loadMonitorDir(QString monitorDir)
     setMonitorDirZims(newDir.values());
     auto manipulator = LibraryManipulator(this);
     auto manager = kiwix::Manager(&manipulator);
+    bool needsRefresh = false;
     for (auto book : addedZims) {
-        manager.addBookFromPath(book.toStdString());
+        needsRefresh = manager.addBookFromPath(book.toStdString()) ? true : needsRefresh;
     }
     for (auto bookPath : removedZims) {
         removeBookFromLibraryById(QString::fromStdString(m_library.getBookByPath(bookPath.toStdString()).getId()));
+        needsRefresh = true;
+    }
+    if (needsRefresh) {
+        emit(booksChanged());
     }
-    emit(booksChanged());
```

Comment:
I wanted to be *sure* to only refresh when a file is actually added.

Consider while downloading a file from Online library from within Kiwix desktop, that creates a `.zim` file while downloading.
Detected by file watcher and is added in `addedZims`, leading to refresh. Just using the condition of `addedZims` being non empty will lead to no use of this fix (this download issue is the main reason)

Though, putting `needsRefresh` in `removedZims` does no good, I'll remove it.

Reply:
> Though, putting needsRefresh in removedZims loop does no good, I'll remove it.

If we remove a book, we want to refresh, so it make sens to have it for me.

Reply:
> If we remove a book, we want to refresh, so it make sens to have it for me.

Not a clear comment by me there, I meant to remove `needsRefresh` from `removedZims` loop and replacing by:
`needsRefresh = removedZims.empty() ? needsRefresh : true;`

That would be okay?

PS. - I edited your comment by mistake with this answer earlier, sorry :)

Reply:
Yes, we agree

Reply:
> I wanted to be _sure_ to only refresh when a file is actually added.
> 
> Consider while downloading a file from Online library from within Kiwix desktop, that creates a `.zim` file while downloading.

That's correct. I had overlooked that `kiwix::Manager::addBookFromPath()` may fail for an invalid ZIM file.

Now the important question is - does `QFileSystemWatcher` **portably** emit `directoryChanged` signal when a file in that directory is modified? If not then this directory monitoring solution may miss download completion events.

Reply:
> Now the important question is - does QFileSystemWatcher portably emit directoryChanged signal when a file in that directory is modified?

Yep it does emit the signal on files modified.

> If not then this directory monitoring solution may miss download completion events.

Even if it did send the signal, it was missing download completion because the file was added using `setMonitorZims()` earlier. To fix that, I moved `setMonitorZims()` inside the `if (needsRefresh)` check.